### ISSUE TITLE
Using SamAutomaticMaskGenerator to predict masks by fine-tuned SAM models but returns an empty list. 

### ISSUE THREAD

**Issue by YunyaGaoTree** [luca-medeiros/lightning-sam#37]
Hello everyone, 

I met a problem when using **SamAutomaticMaskGenerator** to predict masks by using fine-tuned SAM models. 
It always returns an empty list. 
But it works well when following the workflow in the utils.py where the SamPredictor(self.model) is used. 

I used a customized dataset that includes 512x512x3 image patches and json file of annotations. 
The annotations only include two classes that are background and target objects.
I tested the fine-tuned model by the following codes:

from segment_anything import sam_model_registry, SamAutomaticMaskGenerator, SamPredictor
import torch

device = "cuda" if torch.cuda.is_available() else "cpu"
torch.cuda.empty_cache()

model_type = "vit_h"
checkpoint = os.path.join(out_dir, "epoch-000010-f10.84-ckpt.pth")

model = sam_model_registry[model_type](checkpoint=checkpoint)
model.eval()
model.cuda()

image = cv2.imread(path_img)

mask_generator = SamAutomaticMaskGenerator(model)
masks = mask_generator.generate(image)

The models are fine-tuned and tested on Google Colab.

If you happened to solve this issue, look forward to your answers.

**Comment by luca-medeiros** [comment]
As mentioned in the README, this project only supports bounding box input prompts. AutomaticGenerator wouldn't work with the current code

**Comment by YunyaGaoTree** [comment]
> As mentioned in the README, this project only supports bounding box input prompts. AutomaticGenerator wouldn't work with the current code

Thanks so much for your explaination. It is really helpful.

**Comment by RGring** [comment]
If you want to use the SamAutomatucMaskGenerator, you simply need to train with point prompts. I sampled points randomly on the mask. Also make sure that multimask_output=False is set in line 282 if the automatic_mask_generator.py , if you also fine-tune with multimask_output=False.

**Comment by YunyaGaoTree** [comment]
> 

thanks so much for your reply : )
I will try it in the future. 
I found another solution from this link:
https://github.com/tianrun-chen/SAM-Adapter-PyTorch

It is more suitable for semantic segmentation as far as I have tried.

**Comment by kiran-coditation** [comment]
> Hello everyone,
> 
> I met a problem when using **SamAutomaticMaskGenerator** to predict masks by using fine-tuned SAM models. It always returns an empty list. But it works well when following the workflow in the utils.py where the SamPredictor(self.model) is used.
> 
> I used a customized dataset that includes 512x512x3 image patches and json file of annotations. The annotations only include two classes that are background and target objects. I tested the fine-tuned model by the following codes:
> 
> from segment_anything import sam_model_registry, SamAutomaticMaskGenerator, SamPredictor import torch
> 
> ![image](https://user-images.githubusercontent.com/101531630/248787713-b6766ef1-a951-43bf-b871-591c4badded6.png)
> 
> ![image](https://user-images.githubusercontent.com/101531630/248787784-accb2e5d-f8a7-41b8-8f49-7fff68b79ef4.png)
> 
> The models are fine-tuned and tested on Google Colab.
> 
> If you happened to solve this issue, look forward to your answers.

Hi @YunyaGaoTree , I also got the empty list after training for larger epoch. Did you find any solutions for the same?

**Comment by Marchisceddu** [comment]
Hey, my friend and I have been working on a fine-Turing of SAM (automatic predictor) for our thesis. It might be useful for your problem. Check it out https://github.com/Marchisceddu/finestSAM
